packages/visualizations/src/components/utils/ColorsLegend.svelte

```diff
@@ -0,0 +1,117 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+    import type { DataBounds, ColorsScale } from '../types';
+    import { defaultCompactLegendNumberFormat } from './formatter';
+    // options to customize the component
+    export let dataBounds: DataBounds;
+    export let colorsScale: ColorsScale;
+    export let clientWidth: number;
+    export let title: string;
+</script>
+
+<div class="legend-colors {clientWidth <= 375 ? 'legend-colors--fluid' : 'legend-colors--fixed'}">
```

Comment:
I think it should at least be named `parentWidth` or something like that, to make it clear that it is the width of something outside the component.

I'm wondering if it shouldn't rather be up to the component that is using a legend (here, MapRender) to decide whether it want a fixed or fluid legend, using a "variant" property for example. It depends if we consider that the it's up to the legend itself to decide, and that it's always going to be the same rule for every component that uses it.

Reply:
Thanks for your remark ! As the legend component doesn't use the clientWidth (which good point should have been called parentWidth) I applied your second suggestion and passed a variant prop directly from MapRender to the legend component 👍